Library/ViewModels/RewardAddOnSelectionViewModel.swift

```diff
@@ -378,14 +378,14 @@ private func addOnIsAvailable(_ addOn: Reward, in project: Project) -> Bool {
   }
 
   let isUnlimitedOrAvailable = addOn.limit == nil || addOn.remaining ?? 0 > 0
-  let hasNoTimeLimitOrHasNotEnded = (
-    addOn.endsAt == nil ||
-      (addOn.endsAt ?? 0) >= AppEnvironment.current.dateType.init().timeIntervalSince1970
-  )
 
+  // Assuming the user has not backed the addOn, we only display if it's within range of the start and end date
+  let hasNoTimeLimitOrIsWithinRange = isStartDateBeforeToday(for: addOn) &&
+    (addOn.endsAt == nil || (addOn.endsAt ?? 0) >= AppEnvironment.current.dateType.init()
```

Comment:
Should we decouple nil checks and date range check? To make it easier to read.
Also we could create a range 

```
let startDate = Date(timeIntervalSince1970: addOn.startsAt)
let endDate = Date(timeIntervalSince1970: addOn.endsAt)
let range = startDate...endDate

return range.contains(AppEnvironment.current.dateType.init().date)
```

Reply:
and seems we don't even need to convert to `Date`, we should be able to use range from `TimeInterval`

Reply:
I like the idea and syntax of using the range but `RewardsCollectionViewModel` and the `RewardAddOnSelectionViewModel` have slightly different product definitions which is why I created two methods for checking the start and end dates. 

In this case, the `RewardsCollectionViewController` is supposed to hide rewards with a `startsAt` prior to the current date but should show rewards with `endsAt` before today. The `RewardAddOnSelectionViewController`, on the other hand, should be hiding AddOns in both cases (so I use both functions). Let me know if you have another suggestion or optimization.